experiment/popularity_community/crawl_torrents.py

```diff
@@ -0,0 +1,214 @@
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self._output_file_path = out_file_path
+        self.max_peers = UNLIMITED
+        self.queried_peers_limit = UNLIMITED
+
+        self.torrent_list = defaultdict(list)
+        self.polled_peers = set()
+
+        self.register_task("request_torrents", self.request_torrents_from_new_peers, interval=10)
+        self.register_task("save_torrents", self.save, interval=CRAWLER_SAVE_INTERVAL_IN_SEC)
+
+    def request_torrents_from_new_peers(self):
+        """Send a request for gathering torrent information
+
+        Behaviour:
+
+        For all new pears: ```self.get_peers() - self.polled_peers```
+        the crawler request 100 most popular torrents.
+
+        Note:   it is possible to request more, but it requires a significant
+                increase of code complexity.
+        Ref: https://github.com/Tribler/tribler/blob/devel/src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py#L19
+        """
+        peers_ready_to_request = self.get_peers() - self.polled_peers
+        self._logger.info(f"Request torrents from {peers_ready_to_request}")
+
+        for peer in peers_ready_to_request:
+            self._logger.debug(f"Requesting torrents from peer: {peer}")
+            self.send_select(peer, metadata_type=[REGULAR_TORRENT], first=0, last=100)
+            self._logger.debug(f"Marked as polled: {peer}")
+            self.polled_peers.add(peer)
+
+    @lazy_wrapper(SelectResponsePayload)
+    async def on_remote_select_response(self, peer, response):
+        self._logger.debug(f"Received select response from {peer}")
+
+        decompressed_data = lz4.frame.decompress(response.raw_blob)
+        unpacked_data = self.mds.process_squashed_mdblob(decompressed_data)
+
+        for metadata, _ in unpacked_data:
+            if not metadata:
+                continue
+
+            self.add_torrent_to_the_list(peer, metadata)
+
+    def add_torrent_to_the_list(self, peer, metadata):
+        key = hexlify(peer.mid)
+
+        votes = int(getattr(metadata, 'votes', 0))
+        infohash = hexlify(getattr(metadata, 'infohash', ''))
+        title = getattr(metadata, 'title', '')
+
+        self._logger.info(f"Collect torrent item for {key}: {title}")
+        self.torrent_list[key].append((votes, infohash, title))
+
+    # this method has been overloaded only for changing timeout
+    def send_select(self, peer, **kwargs):
+        request = CrawlerSelectRequest(self.request_cache, hexlify(peer.mid), kwargs)
+        self.request_cache.add(request)
+        self.ez_send(peer, RemoteSelectPayload(request.number, json.dumps(kwargs).encode('utf8')))
+
+    def save(self):
+        self._logger.info(f"Save results to : {self._output_file_path}")
+
+        with open(self._output_file_path, 'w') as json_file:
+            json.dump(self.torrent_list, json_file)
+
+    def introduction_response_callback(self, peer, dist, payload):
```

Comment:
Original `introduction_response_callback` triggers `send_remote_select_subscribed_channels`. You don't want that.
So, you have to either stop calling the base class method here, or overload `send_remote_select_subscribed_channels` with empty method, like:

`def send_remote_select_subscribed_channels(...):
    pass
`

Reply:
Fixed